### ISSUE TITLE
Limited fan control on M3(M4) Pro/Max macbooks (display a warning)

### ISSUE THREAD

**Issue by kleuter** [crystalidea/macs-fan-control#785]
On newer Apple Silicon MacBookPro equipped with M3/M4 Pro and Max chips, fan control is heavily dependent on the hardware settings. If the system’s hardware disables fan control, Macs Fan Control (or any other third-party fan control app) won’t be able to override this. This is the reason behind the message you’re seeing and why fan control isn’t available.

In this case the app displays the following warning:

<img width="557" alt="Screenshot 2024-11-20 at 17 48 15" src="https://github.com/user-attachments/assets/a7186515-f7b8-4ba1-9ecc-0a948882374f">

**For those who updated to macOS Sequoia**. Fan control worked for some machines (14inch M3 Pro/Max) on macOS Sonoma but Apple updated their firmware and locked manual fan control in the same way. Once you upgrade to Sequoia there's no way back.

Currently it's confirmed to happen on the following models:

- MacBookPro20,2 - 14inch, Late 2023, M3 Pro [Mac15,6]
- MacBookPro20,3 - 14inch, Late 2023, M3 Max [Mac15,8]
- MacBookPro20,4 - 14inch, Late 2023, M3 Max [Mac15,10]
- MacBookPro20,5 - 16inch, Late 2023, M3 Pro [Mac15,7]
- MacBookPro20,6 - 16inch, Late 2023, M3 Max [Mac15,11]
- MacBookPro20,7 - 16inch, Late 2023, M3 Max [Mac15,9]
- MacBookPro21,2 - 16inch, Late 2024, M4 Pro [Mac16,7]
- MacBookPro21,3 - 16inch, Late 2024, M4 Pro [Mac16,7]
- MacBookPro21,4 - 16inch, Late 2024, M4 Pro [Mac16,7]
- MacBookPro21,5 - 16inch, Late 2024, M4 Max [Mac16,5]

[Download the latest beta](https://github.com/crystalidea/macs-fan-control/issues/721)

[Read full article](https://crystalidea.com/macs-fan-control/limited-fan-control-on-some-models)

**Comment by cigsz** [comment]
"Macs Fan Control **(or any other third-party fan control app)** won’t be able to override this. " well... not entirely true as I was able to control fans however I wanted from Sonoma 14.1.2 till 14.5 (I think? or 14.6) with iStat Menus even when they were turned off.
So some 3rd party apps were able to turn on and control fans, but Apple changed something that now every app is not able to do that.

**Comment by kleuter** [comment]
Once you upgrade to Sequoia there's no way back.

**Comment by valentin56610** [comment]
So, is this the end of MacFansControl?

**Comment by maks2002001** [comment]
[Mac15,6_tech_info.zip](https://github.com/user-attachments/files/17836085/Mac15.6_tech_info.zip)

**Comment by ispaydeu** [comment]
I have never seen the "Manual fan control is restricted on some Apple Silicon-based laptops including 'xxxxxxxx'. It will only take effect once the system activates the fan(s) itself, that is, when it determines that cooling is necessary." on my Macbook Pro Max M3 machine a single time. Both before or after Sequoia. And I have had issues with MacsFanControl working correctly all along on my Macbook Pro Max M3 even before Sequoia and its never displayed this alert. So perhaps you need to fix this alert to ensure it always displays correctly because its not being displayed when it should.

**Comment by maks2002001** [comment]
How to fix this ? 
Z Poważaniem 
Maksymilian Adamczak 

> Wiadomość napisana przez ispaydeu ***@***.***> w dniu 22 lis 2024, o godz. 01:44:
> 
> I have never seen the "Manual fan control is restricted on some Apple Silicon-based laptops including 'xxxxxxxx'. It will only take effect once the system activates the fan(s) itself, that is, when it determines that cooling is necessary." on my Macbook Pro Max M3 machine a single time. Both before or after Sequoia. And I have had issues with MacsFanControl working correctly all along on my Macbook Pro Max M3 even before Sequoia and its never displayed this alert. So perhaps you need to fix this alert to ensure it always displays correctly because its not being displayed when it should.

**Comment by kleuter** [comment]
Users of **MacBookPro18.1**, **MacBookPro18.2** or **MacBookPro18.3** and macOS **Sequoia** 
does manual fan control work for you?

**Comment by maks2002001** [comment]
Nie, ręczne sterowanie nie działa. Cokolwiek się ustawi nie działają wentylatory. 
Z Poważaniem 
Maksymilian Adamczak 

> Wiadomość napisana przez Kleuter ***@***.***> w dniu 22 lis 2024, o godz. 16:22:
> 
> Users of MacBookPro18.1, MacBookPro18.2 or MacBookPro18.3 and macOS Sequoia
> does manual fan control work for you?
>

**Comment by kleuter** [comment]
Can anyone confirm that? The mac should be running macOS **Sequoia**

**Comment by maks2002001** [comment]
Confirm what ? Z Poważaniem Maksymilian Adamczak Wiadomość napisana przez Kleuter ***@***.***> w dniu 22 lis 2024, o godz. 16:42:﻿
Can anyone confirm that? The mac should be running macOS Sequoia

**Comment by cigsz** [comment]
To some people with M3 Macs - I have downgraded macOS Sonoma from 14.7.1 to 14.3.1 and I'm able to once again control my fans however I want, which confirms that Apple intentionally blocked it with an update. 
So if you want to have control over your fans again - you have to downgrade or wait and pray that Apple enable it in future update.

**Comment by kleuter** [comment]
@cigsz which macbook model do you have, please?

**Comment by maks2002001** [comment]
MacBook Pro 14” M3 Pro 
Z Poważaniem 
Maksymilian Adamczak

**Comment by cigsz** [comment]
> @cigsz which macbook model do you have, please?

MacBookPro20,2 - 14inch, Late 2023, M3 Pro [Mac15,6]

**Comment by maks2002001** [comment]
MRX33ZE/A
Z Poważaniem 
Maksymilian Adamczak

**Comment by omar-toma** [comment]
> Help wanted!
> 
> Users of **MacBookPro18.1**, **MacBookPro18.2** or **MacBookPro18.3** and macOS **Sequoia** does manual fan control work for you?

doesn't for me on my m3 pro 15.1.1 (24B91)

**Comment by weidali** [comment]
Got it, thanks for the explanation! 
It's a shame that Apple locked manual fan control in Sequoia. Looks like we'll have to live with the firmware restrictions for now. 

Hopefully, there will be solutions or updates in the future that bring this feature back. For now, I guess we can only hope for better fan management in macOS itself.

**Comment by westito** [comment]
I have a manual "workaround" that could be done in the app. I run this GPU stress test continously: https://www.matthew-x83.com/online/gpu-test.php
Thats increases the temperature engouh to start up fans, but not affects CPU performance. I think a small routine could be bundled in the app that runs some GPU computing task in the background.

**Comment by jadenquinn** [comment]
> I have a manual "workaround" that could be done in the app. I run this GPU stress test continously: https://www.matthew-x83.com/online/gpu-test.php Thats increases the temperature engouh to start up fans, but not affects CPU performance. I think a small routine could be bundled in the app that runs some GPU computing task in the background.

That's... rough..

**Comment by vlbosch** [comment]
Yesterday an update was released for TG Pro (Tunabelly) which regains control over the fans, even when the system has turned them off. I can confirm it works on M3 Max, running macOS 15.3.1. However, I didn't find a release blog yet, only noticed it via the changelog. Maybe their workaround can be copied @kleuter?

**Comment by sm1smbdy** [comment]
> Yesterday an update was released for TG Pro (Tunabelly) which regains control over the fans, even when the system has turned them off. I can confirm it works on M3 Max, running macOS 15.3.1. However, I didn't find a release blog yet, only noticed it via the changelog. Maybe their workaround can be copied @kleuter?

Sadly it's not working on my M3 Pro MBP model 16'' (Sonoma 14.5)... According to the change log the fix with manual fan control was related to m4 family chips primarily, but somehow, from your words, m3 models could be related there as well.

**Comment by kleuter** [comment]
Indeed, some improvement can be done here. 

Our workaround is a little different, a beta build will be soon

**Comment by kleuter** [comment]
https://github.com/crystalidea/macs-fan-control/issues/822

**Comment by kleuter** [comment]
Use beta to test, please

**Comment by westito** [comment]
Nice job! It's working on my Apple M4 Max 
Can you share some details? I'm very curious about the solution (hidden option, fake sensor values, code inject, etc.)

**Comment by westito** [comment]
However, the fans not turning off now, even the temperature much lower than the set value. Stays az 1300rpm. Is this the expected behavior by the workaround?

**Comment by kleuter** [comment]
If you change back to Auto - it will disable when T is low.

We do not currently share any information on Macs Fan Control tricks 😉

**Comment by westito** [comment]
Yeah it's okay. This will be fine-tuned later? Ex. changes back automatically to Auto when temperature is low.

**Comment by kleuter** [comment]
I didn't disable the notification yet but the fan(s) should be turning on.

**Please let me know how much actual time passes before fans turns on!**

**Comment by IgnacioEngelberger** [comment]
It is working on my MacBook M3 pro as well 🎉

**Comment by shahmir-oscilar** [comment]
> **Please let me know how much actual time passes before fans turns on!**

Worked great. Took about 3-4 seconds before the fans turned on; totally acceptable in my book.

**Comment by jadenquinn** [comment]
> > **Please let me know how much actual time passes before fans turns on!**
> 
> Worked great on my M4 Pro Macbook Pro. Took about 3-4 seconds before the fans turned on; totally acceptable in my book.

Doesn't work on mine. Still have to wait for the die to heat up for the fans to come on themselves, and THEN I can control them. Same behavior since the beginning.

**Comment by jadenquinn** [comment]
Oh I found it I think. The 1.5.18 one. Yea about 4-5 seconds until it turns the fans on manually! Legendary.
